sklearn/decomposition/_dict_learning.py

```diff
@@ -1184,19 +1184,15 @@ class DictionaryLearning(_BaseSparseCoding, BaseEstimator):
 
     transform_n_nonzero_coefs : int, default=None
         Number of nonzero coefficients to target in each column of the
-        solution. This is only used by `algorithm='lars'` and `algorithm='omp'`
-        and is overridden by `alpha` in the `omp` case. If `None`, then
-        `transform_n_nonzero_coefs=int(n_features / 10)`.
+        solution. This is only used by `algorithm='lars'` and `algorithm='omp'`.
+        If `None`, then `transform_n_nonzero_coefs=int(n_features / 10)`.
 
     transform_alpha : float, default=None
         If `algorithm='lasso_lars'` or `algorithm='lasso_cd'`, `alpha` is the
         penalty applied to the L1 norm.
         If `algorithm='threshold'`, `alpha` is the absolute value of the
         threshold below which coefficients will be squashed to zero.
-        If `algorithm='omp'`, `alpha` is the tolerance parameter: the value of
-        the reconstruction error targeted. In this case, it overrides
-        `n_nonzero_coefs`.
-        If `None`, default to 1.0
+        If `None`, default to `alpha`.
```

Comment:
```suggestion
        If `None`, defaults to `alpha`.
```

Reply:
This breaks backward compat. We probably need to do a deprecation cycle (https://scikit-learn.org/stable/developers/contributing.html#maintaining-backwards-compatibility)

I think we should issue a warning when alpha is not left to it's default value but transform_alpha is.